src/providers/arcgisrest/qgsarcgisrestutils.cpp

```diff
@@ -964,6 +982,24 @@ QgsFeatureRenderer *QgsArcGisRestUtils::parseEsriRenderer( const QVariantMap &re
   return nullptr;
 }
 
+QString QgsArcGisRestUtils::parseEsriLabelingExpression( const QString &string )
+{
+  QString expression = string;
+
+  // Replace a few ArcGIS token to QGIS equivalents
+  expression = expression.replace( QRegularExpression( "(?=([^\"\\\\]*(\\\\.|\"([^\"\\\\]*\\\\.)*[^\"\\\\]*\"))*[^\"]*$)(\\s|^)CONCAT(\\s|$)" ), QStringLiteral( "\\4||\\5" ) );
```

Comment:
@nirvn  Sorry to jump in late, just a minor recommendation: in order to make the regexp more readable you could use the raw string syntax and skip all that \\ escaping, something like this:
```c++
expression = expression.replace( QRegularExpression( R"re((?=([^"\\]*(\.|"([^"\\]*\.)*[^"\]*"))*[^"]*$)(\s|^)CONCAT(\s|$))re" ), QStringLiteral( R"re(\4||\5)re" ) );
```